mobile/ios/WirelessDebug/WirelessDebugger/WebSocketMessenger.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  WebSocketMessenger.swift
+//  Wireless Debug iOS Test App
+//
+
+import UIKit
+import Foundation
+
+/// Handles opening a WebSocket connection and sending the necessary messages
+/// across that connection.
+class WebSocketMessenger {
+    var webSocket = WebSocket()
+    var logsToSend: [String] = []
+    var connectionRetries = 0
+    var verbose: Bool
+
+    /// Create a WebSocket connection to the given socket address and API Key.
+    ///
+    /// - Parameters:
+    ///   - socketAddress: the WebSocket address to connect to
+    ///   - apiKey: the API Key provided by the web interface
+    init(_ socketAddress: String, apiKey: String, verbose: Bool) {
+        self.verbose = verbose
+        // When the WebSocket connection opens, send a startSession message.
+        self.webSocket.event.open = {
+            self.log("WebSocket Connection to \(socketAddress) Opened")
+
+            self.send("startSession", data: [
+                "osType": "iOS",
+                "apiKey": apiKey,
+                "deviceName": UIDevice.current.name,
+                "appName": Bundle.main.infoDictionary!["CFBundleName"] as! String,
+            ])
+            self.connectionRetries = 0
+        }
+
+        // If the connection closes, try to reconnect 10 times.
+        self.webSocket.event.close = { code, reason, clean in
+            self.log("WebSocket Connection Closed \(reason)")
+
+            // Retry the connection 10 times with 2 seconds inbetween retries.
+            if self.connectionRetries < 10 {
+                DispatchQueue.main.asyncAfter(deadline: .now() + 2.0) {
+                    self.webSocket.open(socketAddress)
+                }
+                self.connectionRetries += 1
+            }
+        }
+
+        self.webSocket.event.error = { error in
+            self.log("Error on WebSocket Connection \(error)")
+        }
+
+        // Now actually try to open the WebSocket connection.
+        self.webSocket.open(socketAddress)
+    }
+
+    /// Send the queued logs across the WebSocket connection.
+    public func sendLogDump() {
```

Comment:
I understand, my comment was to shed light on a path that would cut out ~30% of the code.

Currently - LogStreamer has two tasks that run "independently" of each other. Read/Send. Read populates a queue on WebSocketMessenger. Send notifies WebSocketMessenger to upload the cached data. For read, it blocks the queue until there is available data. For sends, it sends whenever a read happens due to the block.

Proposal - LogStreamer has one task that reads the current buffer, and sends the payload over. This would simplify a lot of logic and grants the same functionality. It cuts two tasks out, removes the need of a queue on the client or logstreamer, lowers coupling from logstreamer <-> client that allows for swapping out clients if websockets is not suitable. 

example:
In logstreamer:
ProcessLogs {
  payload = ReadPipe() //(blocks)
  status = client.SendPayload(payload) //(synchronous)
  if status = fail { stop } 
  else {  dispatch(ProcessLogs, 50ms)  }
  }

In client: 
SendPayload(Data) {
  // schema declaration
  socket.Send(data)
  // error handling
  return status
}